GoogleContainerTools/skaffold
# feat: crd status check

Fixes: #8462
**Description**
 - users can provide resource selectors through file for their crds status check 


**Test Plan**
  
 - I don't know any other crd controllers than ConfigConnector that implements kstatus, you can test other crds if you know those 
 - checkout this branch , comment out this line https://github.com/GoogleContainerTools/skaffold/blob/700fd81d5ee28977c1bc0da6c34336925cdc205b/pkg/skaffold/kubernetes/manifest/selector.go#L62 (otherwise, the code will not use customize resource validator) and build skaffold
 - follow this guide to set up Config Connector https://cloud.google.com/config-connector/docs/how-to/getting-started
 -  download this tar
[test-crds-status-check.tar.gz](https://github.com/GoogleContainerTools/skaffold/files/12323816/test-crds-status-check.tar.gz) and untar it 
 - enter the result folder and run `skaffold dev --status-check-selectors selectors.json` should see status check against the crd, the status check against ConfigConnector will hang until check success, this is expected as config controller returns IN_PROGRESS for both resource reconciling and failing.
